exiv2: build failure on aarch64-darwin since 2021

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix build nixpkgs#exiv2`

### Build log

```
======================================================================
FAIL: test_run (bugfixes.redmine.test_issue_683.PercentABrokenInRename)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-exiv2-0.27.5.drv-0/source/tests/bugfixes/redmine/test_issue_683.py", line 26, in tearDown
    self.assertTrue(os.path.exists(self.new_file))
AssertionError: False is not true

----------------------------------------------------------------------
Ran 266 tests in 22.655s
```

See [Hydra](https://hydra.nixos.org/job/nixpkgs/staging-next/exiv2.aarch64-darwin/all?page=6).

### Additional context

- Adding `rm -f ../tests/bugfixes/redmine/test_issue_683.py` in `preCheck` fixes the build on my machine.
- Broken since #127172.

### Notify maintainers

@paperdigits

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

--- a/pkgs/development/libraries/exiv2/default.nix
+++ b/pkgs/development/libraries/exiv2/default.nix
@@ -87,6 +87,8 @@ stdenv.mkDerivation rec {
       substituteInPlace  ../tests/bash_tests/testcases.py \
         --replace "def io_test(self):" "def io_disabled(self):"
      ''}
+  '' + lib.optionalString (stdenv.isDarwin && stdenv.isAarch64) ''
+    export LC_ALL=C
   '';
 
   # With CMake we have to enable samples or there won't be
@@ -120,6 +122,6 @@ stdenv.mkDerivation rec {
     description = "A library and command-line utility to manage image metadata";
     platforms = platforms.all;
     license = licenses.gpl2Plus;
-    maintainers = [ ];
+    maintainers = with maintainers; [ wegank ];
   };
 }

